lib/swm.js

```diff
@@ -114,6 +124,15 @@ export class Client {
     this.sendMessage(this.createMessage(messageType, payload));
   }
 
+  async getMessageResponse(messageType, payload={}) {
+    this.inflight.promise = new Promise(function (resolve, reject) {
+      this.inflight.resolve = resolve;
+      this.inflight.reject = reject;
+      this.send(messageType, payload);
```

Comment:
I suppose I should add a timeout, here, too and reject the promise if there's no response within 800ms or something.

Reply:
Leaving this for now, but it might help to have an optional timeout parameter to the function, which could be used in a setTimeout call.